jump-core/src/main/java/com/bitdecay/jump/BitWorld.java

```diff
@@ -321,12 +454,15 @@ public BitPointInt getBodyOffset() {
 	}
 
 	public void setObjects(Collection<BitBody> otherObjects) {
-		pendingRemoves.addAll(bodies);
+		pendingRemoves.addAll(dynamicBodies);
+		pendingRemoves.addAll(kineticBodies);
+		pendingRemoves.addAll(staticBodies);
+
 		pendingAdds.addAll(otherObjects);
 	}
 
 	public void removeAllBodies() {
-		pendingRemoves.addAll(bodies);
+		pendingRemoves.addAll(dynamicBodies);
```

Comment:
Shouldn't this include static and kinetic bodies? Or should the method be "removeAllDynamicBodies"?